### PR TITLE
Dummy chat messages in encyclopedia

### PR DESCRIPTION
**Pull request by ItsSammyM**

### CODE REVIEW

File: client/src/components/ChatMessage.tsx [hunk 1]
```diff
@@ -10,8 +10,17 @@ import { Grave } from "../game/graveState";
...
+export default function ChatElement(
+    props: {
+        message: ChatMessage,
+        playerNames?: string[]
+    }, 
+): ReactElement {
     const message = props.message;
+    if (props.playerNames === undefined) {
+        props.playerNames = getPlayerNames();
+    }
```
Comment (Jack-Papel) [inline client/src/components/ChatMessage.tsx:22]: God is good

File: client/src/components/ChatMessage.tsx [hunk 2]
```diff
@@ -66,16 +75,22 @@ export default function ChatElement(props: {message: ChatMessage}): ReactElement
...
+export function getPlayerNames(): string[] {
+    switch (GAME_MANAGER.state.stateType) {
+        case "game":
+            return GAME_MANAGER.state.players.map((player) => player.toString());
+        default:
+            return [];
+    }
+}
+
+function playerListToString(playerList: PlayerIndex[], playerNames: string[]): string {
```
Comment (Jack-Papel) [inline client/src/components/ChatMessage.tsx:90]: nit: this function doesn't take a player list as input, should rename?

File: client/src/components/ChatMessage.tsx [hunk 3]
```diff
@@ -85,44 +100,43 @@ export function sanitizePlayerMessage(text: string): string {
     });
 }
 
-export function translateChatMessage(message: ChatMessage): string {
-    if(GAME_MANAGER.state.stateType !== "game"){
-        return "ERROR: chatmessages cant exist outside game"
-    }
+export function translateChatMessage(message: ChatMessage, playerNames: string[]): string {
```
Comment (Jack-Papel) [inline client/src/components/ChatMessage.tsx:103]: that second parameter can definitely be inferred if absent IMO. I think ChatMenu uses function this for chat filtering

File: client/src/components/StyledText.tsx [hunk 4]
```diff
@@ -39,7 +40,14 @@ type Token = {
  * 
  * @see sanitizePlayerMessage in ChatMessage.tsx
  */
-export default function StyledText(props: { children: string[] | string, className?: string, noLinks?: boolean, markdown?: boolean }): ReactElement {
+export default function StyledText(props: {
+        children: string[] | string,
+        className?: string,
+        noLinks?: boolean,
+        markdown?: boolean,
+        playerNames?: string[]
```
Comment (Jack-Papel) [inline client/src/components/StyledText.tsx:48]: This parameter is unused. Should be removed
```suggestion
```

File: client/src/components/StyledText.tsx [hunk 5]
```diff
@@ -97,6 +105,26 @@ function clearKeywordData() {
...
+            { replacement: " " },
+            { style: "keyword-player-sender", replacement: name }
+        ];
+        KEYWORD_DATA_MAP[name] = [
+            { style: "keyword-player-number", replacement: (i + 1).toString() },
+            { replacement: " " },
+            { style: "keyword-player", replacement: name }
+        ];
+    }
+
```
Comment (Jack-Papel) [inline client/src/components/StyledText.tsx:127]: Lol, so I can send 1: Cotton Mather and it'll render? Kinda funny. Would be nice if there was a way to limit these dummy names to just the wiki tho...
I'm fine with merging like this though. Can be a TODO for later.
Add a TODO here for me

File: client/src/components/WikiArticle.tsx [hunk 6]
```diff
@@ -38,21 +40,36 @@ export default function WikiArticle(props: {
...
+                    {(translateChecked("wiki.article.role."+role+".extra") ?? translate("wiki.article.role.noExtra"))+"\n"}
+
+                    {"### "+translate("wiki.article.standard.roleLimit.title")+": "+(roleData.maxCount === null ? translate("none") : roleData.maxCount)+"\n"}
+                    {"### "+translate("defense")+": "+translate("defense."+(roleData.armor ? "1" : "0"))+"\n"}
+
+                    {"### "+translate("wiki.article.standard.chat.title")+"\n"}
+                </StyledText>
+                <div className="wiki-message-section">
+                    {roleData.chatMessages.map((msg: any)=>
+                        <><ChatElement message={msg as ChatMessage} playerNames={DUMMY_NAMES}/></>
```
Comment (Jack-Papel) [inline client/src/components/WikiArticle.tsx:67]: Is this fragment necessary?

File: client/src/resources/roles.json [hunk 7]
```diff
@@ -5,263 +5,487 @@
...
+        "canBeConvertedTo": [],
+        "chatMessages":[
+            {"type": "youAreInterviewingPlayer","playerIndex": 1},
+            {"type": "playerIsBeingInterviewed","playerIndex": 1},
+            {
+                "type": "normal", 
+                "messageSender": {
+                    "type": "journalist"
+                },
+                "text": "Could you give us your opinion on the situation happening in [REDACTED]?",
```
Comment (Jack-Papel) [inline client/src/resources/roles.json:288]: This is funny, but people might think [REDACTED] has some special meaning.
Consider:
image
Making medium this ^ and journalist the football one

**Review by Jack-Papel — CHANGES_REQUESTED** [review]

File: client/src/components/ChatMessage.tsx (hunk 2 above)
Comment (Jack-Papel) [reply client/src/components/ChatMessage.tsx:90]: `playersToString`?

Comment (Jack-Papel) [reply client/src/components/ChatMessage.tsx:90]: `translatePlayers`?

File: client/src/components/ChatMessage.tsx (hunk 3 above)
Comment (ItsSammyM) [reply client/src/components/ChatMessage.tsx:103]: I wanted to do that.

It needs to check this however
gamemanager.state.statetype === "game"
I dont want to have it set to undefined or something.

File: client/src/components/WikiArticle.tsx (hunk 6 above)
Comment (ItsSammyM) [reply client/src/components/WikiArticle.tsx:67]: we cant put that div inside style text

File: client/src/components/ChatMessage.tsx (hunk 3 above)
Comment (Jack-Papel) [reply client/src/components/ChatMessage.tsx:103]: On line 20, you do 
```tsx
    if (props.playerNames === undefined) {
        props.playerNames = getPlayerNames();
    }
```

Comment (Jack-Papel) [reply client/src/components/ChatMessage.tsx:103]: It would be a good idea to do that here, perhaps

**Comment by Jack-Papel** [comment]
Closes #454

File: client/src/components/ChatMessage.tsx (hunk 2 above)
Comment (ItsSammyM) [reply client/src/components/ChatMessage.tsx:90]: That is a list of players

File: client/src/components/StyledText.tsx (hunk 5 above)
Comment (ItsSammyM) [reply client/src/components/StyledText.tsx:126]: no, make that an issue

**Review by Jack-Papel — APPROVED** [review]

**Pull request merged**